app/routes/circulars/circulars.server.ts

```diff
@@ -172,3 +173,13 @@ export async function put(subject: string, body: string, request: Request) {
     submitter: formatAuthor(user),
   })
 }
+
+export async function circularRedirect(query: string | undefined) {
+  if (!query) return
+  const validCircularSearchStyles = /(GCN\s*|Circular\s*|(^[0-9]+$))/i
+  if (validCircularSearchStyles.test(query)) {
+    const circularId = query.replace(/\D/g, '')
```

Comment:
In your regex, having surrounded the actual circular number in a capturing group, you can simplify the test for a matching circular ID using optional chaining like this:

```ts
const circularId = validCircularSearchStyles.exec(query)?.[0]
if (circularId) {
   // do redirect
}
```